### ISSUE TITLE
VM + Intro to VM tutorial

### ISSUE THREAD

**Issue by barisbalic** [codebar/tutorials#131]
There's obviously a problem in the variety of systems and the configuration of the systems that are brought along to codebar.  It means people have inconsistent experiences and tutorials have to explain up to three ways of doing things when it's OS dependent.

I propose (and will begin) creating a VM that can be shared with students and made available online, and an accompanying tutorial about using the VM.  Basic tutorial should include:
- What a VM is.
- How to install software in ubuntu.
- Some analogs for well known software.
- Super basic command line usage.
- Explanation of `sudo` and why exists.

Suggestions/feedback?

**Comment by deniseyu** [comment]
+1 for command line basics. I'd be willing to put together a quick tutorial on using the command line for version control and maybe (a teeny tiny bit) of text-editing in Vim

**Comment by AVGP** [comment]
I'd be in for the VM creation for this.

**Comment by barisbalic** [comment]
I've been drafting an overall CLI/shell tutorial, but it keeps growing, would be good to have feedback on it, will start another thread.

**Comment by matyikriszta** [comment]
@barisbalic @AVGP @deniseyu these are all good ideas. I'd say let's separate out the VM creation and CLI/shell tutorial into two separate issue. @barisbalic if you want to share what you've put together so far I'm happy to look at it and make suggestions. @deniseyu maybe you and @barisbalic could collaborate on this?

**Comment by barisbalic** [comment]
@matyikriszta yes that was always the intention.  CLI was meant to be 'enough CLI to install things in the VM', but it blew up a bit.  @deniseyu I would be happy for the assist if you are game.

**Comment by deniseyu** [comment]
@barisbalic @matyikriszta would love to help with command line. never had to use a VM before so i'm probably not going to be too useful on that!

**Comment by matyikriszta** [comment]
hey @barisbalic & @deniseyu, can I put a deadline on this? Say end of this month the 29th of March?

**Comment by barisbalic** [comment]
@matyikriszta @deniseyu I would like to commit to that deadline but time is escaping me this month.  It may be early to mid april :(

**Comment by alexanderkustov** [comment]
https://nanobox.io/

This has launched recently, and looks like a potential simplified method of understanding what a VM is...
